packages/devtools_app/lib/src/analytics/_analytics_web.dart

```diff
@@ -689,11 +689,19 @@ Map<String, dynamic> generateSurveyQueryParameters() {
   final url = window.location.toString();
   const fromValuePrefix = '#/';
   final startIndex = url.indexOf(fromValuePrefix);
-  final endIndex = url.indexOf('?');
-  final fromPage = url.substring(
-    startIndex + fromValuePrefix.length,
-    endIndex,
-  );
+  // Use the last index because the url can be of the form
+  // 'http://127.0.0.1:9103/?#/?' and we want to be referencing the last '?'
+  // character.
+  final endIndex = url.lastIndexOf('?');
+  var fromPage = '';
+  try {
```

Comment:
probably check that endIndex !=-1 before this to avoid having to throw an exceptions.